### PR TITLE
Copy content from txt file to another

### LINKED ISSUE #22: Copy Content from One .txt File to Another
Params: source_file (str) , Destination_file(str) 
Return : None-The function performs the copying operation but does not return anything

### PR DESCRIPTION
**Pull request by sajaabusafia**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by Safaishag** [comment]
Dear, Saja
Your work is perfect and complete, it is passing all the CI checks and the Review checks as well. Below are my suggestions but not necessary, you are free to decide!

Expand Test Cases:
Add tests for edge cases such as:

       Source file with special characters in the name.

        File paths in nested directories.

        Destination file with restricted permissions.

File: solutions/tests/test_copy_from_file_to_file.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            self.assertEqual(source.read(), destination.read())
+
+    def test_src_not_exist(self):
+        """Test when the source file does not exist"""
+        source_file = "solutions/not_exist.txt"
+        destination_file = "solutions/destination.txt"
+
+        # Check if it raises an assertion error when the source does not exist
+        with self.assertRaises(AssertionError):
+            copy_file_to_file(source_file, destination_file)
```
Comment (Safaishag) [inline solutions/tests/test_copy_from_file_to_file.py:85]: Dear, Saja
Your work is perfect and complete, it is passing all the CI checks and the Review checks as well. Below are my suggestions but not necessary, you are free to decide!

Expand Test Cases:
Add tests for edge cases such as:

   Source file with special characters in the name.

    File paths in nested directories.

    Destination file with restricted permissions.

**Review by Safaishag — CHANGES_REQUESTED** [review]

File: solutions/tests/test_copy_from_file_to_file.py (hunk 1 above)
Comment (Safaishag) [reply solutions/tests/test_copy_from_file_to_file.py:85]: well done!

**Review by Safaishag — APPROVED** [review]
well done!

File: solutions/tests/test_copy_from_file_to_file.py (hunk 1 above)
Comment (sajaabusafia) [reply solutions/tests/test_copy_from_file_to_file.py:85]: Thank you, Safa, for your suggestions! I honestly hadn’t thought of that.
